crawler/file_processing.py

```diff
@@ -493,6 +501,81 @@ def add_duplication_errors(self, exception):
         except Exception as e:
             logger.critical(f"Unknown error with file {self.file_name}: {e}")
 
+    def docs_to_insert_updated_with_source_plate_uuids(
+        self, docs_to_insert: List[Dict[str, Any]]
+    ) -> List[Dict[str, Any]]:
+        """Updates sample records with source plate uuids, returning only those for which a source
+        plate uuid could be determined. Adds any new source plates to mongo.
+
+        Arguments:
+            docs_to_insert {List[Dict[str, Any]]} -- the sample records to update
+
+        Returns:
+            List[Dict[str, Any]] -- the updated, filtered samples
+        """
+        logger.debug("Attempting to update docs with source plate uuids")
+        updated_docs = []
+
+        def update_doc_from_source_plate(doc, existing_plate, skip_lab_check=False) -> None:
```

Comment:
Just curious about the nested def for python - I haven't seen it before. Looks great just what are the advantages doing it this way, as oppose to outside the method definition? Could you point me to any documentation about nested defs

Reply:
Also one of my first uses of this workflow. I did it this way as you can access properties defined in the parent method within this one, rather than declaring new class variables (`updated_docs`). I guess it also guarantees it won't be called outside of the scope it was intended for

Reply:
Ah nice! Ye really cool